### PR TITLE
feat: gbp usd tracking

### PR DESCRIPTION
**Pull request by jmulq**
The purpose of these changes, is so that we can keep track of the amount of currency spent to fill a listing.

The total amount is the set value, the fees are the total minus the 1.5% fee paid by the purchaser of the NFT, and the amount minus fees are the remainder that goes to the listing creator. It is required that we keep a GBP/USD conversion of our sales.

### CODE REVIEW

File: src/graphql-types/bid.ts [hunk 1]
```diff
@@ -27,6 +27,14 @@ export class Bid {
   @property()
   amount: Price
 
+  @field((type) => Price)
```
Comment (anjorinjnr) [inline src/graphql-types/bid.ts:30]: are there fees on bid?

File: src/graphql-types/fillPriceData.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { field, inputType, objectType } from '@loopback/graphql'
+import { property } from '@loopback/repository'
+import { Price } from './price'
+
+@inputType('PriceInput')
+@objectType({ description: 'Amount for a asset' })
```
Comment (anjorinjnr) [inline src/graphql-types/fillPriceData.ts:6]: what is this object for?

File: src/graphql-types/bid.ts (hunk 1 above)
Comment (jmulq) [reply src/graphql-types/bid.ts:30]: yeah - in the sense that a bid is just a 0x order waiting to be filled, and the bidder will pay a fee of 1.5% to VH, similar to a buy now listing.

**Comment by jmulq** [comment]
> you can just have a fillAmount field in the listing object
> 
> ```ts
> {
>   amount: Price
>   fee: Price
>   netAmount: Price
>   usdRate: number
>   gbpRate: number
> }
> ```

Where net amount is the amount minus fees? Or with fees (i.e. the total)

Is the usdRate and gbpRate just the rate of the chosen currency at the time? So 1 VHC in USD/GBP, and then we can do any calculations we need to at a later date?

File: src/graphql-types/bid.ts (hunk 1 above)
Comment (anjorinjnr) [reply src/graphql-types/bid.ts:30]: but why do you need the information on the bid?
The fees only apply when  the bid is accepted and a txn is created

File: src/services/currency-exchange.service.ts/currency-exchange.service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { BindingScope, injectable } from '@loopback/core'
+import fetch from 'node-fetch'
+
+@injectable({ scope: BindingScope.TRANSIENT })
```
Comment (anjorinjnr) [inline src/services/currency-exchange.service.ts/currency-exchange.service.ts:4]: you really could just make this a simple function

File: src/graphql-types/listing.ts [hunk 4]
```diff
@@ -90,9 +91,9 @@ export class FillDetails {
   @property()
   takerAddress: string
 
-  @field((type) => Price)
+  @field((type) => FillAmount)
   @property()
-  price: Price
+  priceData: FillAmount
```
[the remark this answers is not in the document]
Comment (jmulq) [reply src/graphql-types/listing.ts]: Good point. Have updated this in latest changes.

File: src/services/currency-exchange.service.ts/currency-exchange.service.ts (hunk 3 above)
Comment (jmulq) [reply src/services/currency-exchange.service.ts/currency-exchange.service.ts]: Made into a helper func in changes

File: .github/workflows/deploy-prod.yaml [hunk 5]
```diff
@@ -10,12 +10,15 @@ env:
   EBS_APP_NAME: vh-marketplace-server-prod
   EBS_ENVIRONMENT_NAME: vh-marketplace-server-prod-env
   S3_BUCKET: vh-marketplace-server-prod
-  ALCHEMY_API_KEY: ${{ secrets.ALCHEMY_API_KEY }}
+  ALCHEMY_API_KEY: ${{ secrets.ALCHEMY_API_PROD }}
   CHAIN_ID: 3
   DB_URL: ${{ secrets.MARKETPLACE_PROD_DB_URL }}
```
Comment (anjorinjnr) [inline .github/workflows/deploy-prod.yaml:15]: change to MARKETPLACE_DB_URL_PROD

File: .github/workflows/deploy-prod.yaml [hunk 6]
```diff
@@ -10,12 +10,15 @@ env:
...
-  PRIVATE_KEY: ${{ secrets.MARKETPLACE_PROD_RIVATE_KEY }}
+  FEE_PERCENTAGE: 1.5
+  FEE_RECIPIENT: '0x565eCe738084b8cFdD20f3aA97dB1fA3Eafd24B9'
+  PRIVATE_KEY: ${{ secrets.MKT_PRIVATE_KEY }}
   SUBGRAPH_API_URL: ${{ secrets.PROD_SUBGRAPH_API }}
```
Comment (anjorinjnr) [inline .github/workflows/deploy-prod.yaml:19]: change to SUBGRAPH_API_PROD

File: .github/workflows/deploy-prod.yaml [hunk 7]
```diff
@@ -10,12 +10,15 @@ env:
...
   EBS_ENVIRONMENT_NAME: vh-marketplace-server-prod-env
   S3_BUCKET: vh-marketplace-server-prod
-  ALCHEMY_API_KEY: ${{ secrets.ALCHEMY_API_KEY }}
+  ALCHEMY_API_KEY: ${{ secrets.ALCHEMY_API_PROD }}
   CHAIN_ID: 3
   DB_URL: ${{ secrets.MARKETPLACE_PROD_DB_URL }}
-  PRIVATE_KEY: ${{ secrets.MARKETPLACE_PROD_RIVATE_KEY }}
+  FEE_PERCENTAGE: 1.5
+  FEE_RECIPIENT: '0x565eCe738084b8cFdD20f3aA97dB1fA3Eafd24B9'
+  PRIVATE_KEY: ${{ secrets.MKT_PRIVATE_KEY }}
```
Comment (anjorinjnr) [inline .github/workflows/deploy-prod.yaml:18]: MARKETPLACE_PRIVATE_KEY_PROD

**Review by anjorinjnr — CHANGES_REQUESTED** [review]
naming consistency

**Comment by jmulq** [comment]
> naming consistency

Just updated these following comments

**Review by anjorinjnr — APPROVED** [review]

**Pull request merged**